pkg/node/store/store.go

```diff
@@ -81,16 +87,27 @@ func (o *NodeObserver) OnDelete(k store.NamedKey) {
 	if n, ok := k.(*node.Node); ok {
 		nodeCopy := n.DeepCopy()
 		nodeCopy.Source = node.FromKVStore
-		o.manager.NodeDeleted(*nodeCopy)
 
-		ciliumIPv4 := nodeCopy.GetCiliumInternalIP(false)
-		if ciliumIPv4 != nil {
-			ipcache.IPIdentityCache.Delete(ciliumIPv4.String(), ipcache.FromKVStore)
-		}
-		ciliumIPv6 := nodeCopy.GetCiliumInternalIP(true)
-		if ciliumIPv6 != nil {
-			ipcache.IPIdentityCache.Delete(ciliumIPv6.String(), ipcache.FromKVStore)
-		}
+		func() {
```

Comment:
Is this intended to be `go func() {`?

Otherwise the synchronous sleep below seems like it may cause trouble if many node delete events arrive at once.

Reply:
+1, I would expect a goroutine as well.